### PR TITLE
README v1

### PR DESCRIPTION
**Pull request by nicandris**
Hey @andrei-tatar I added a small readme. Can you review it and publish/reject? There are some comments in there from me that i need you to clarify. Comment in the PR and I'll fix whatever is wrong and push again.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,103 @@
...
+- Create a project
+- Choose 'Smart Home'
+- Name your Action
+- Go back to Overview and Setup Account Linking 
+	- Linking Type: OAuth - Authorization Code
+	- Client ID: this is your Client Id from Google API Console ('OAuth 2.0 Client IDs' click your webclient credentials)
+	- Client secret: this is your Client Secret from Google API Console ('OAuth 2.0 Client IDs' click your webclient credentials)
+	- Authorization URL - https://XXXXXXXXX.herokuapp.com/oauth
+	- Token URL - https://XXXXXXXXX.herokuapp.com/oauth/token
+	- Scopes (I added google-home-authcode / app-user because i saw those 2 somewhere in the code but I don't think they are needed) ?? @Andrei please confirm/deny
```
Comment (andrei-tatar) [inline README.md]: No scopes are needed here.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,103 @@
...
+    databaseURL: 'firebase - settings -general - your apps',
+    projectId: 'firebase - settings -general - your apps',
+    storageBucket: 'firebase - settings -general - your apps',
+    messagingSenderId: 'firebase - settings -general - your apps',
+
+
+## USER.REPOSITORY.TS
+Modify [src/services/user.repository.ts](https://github.com/andrei-tatar/nora-service/blob/master/src/services/user.repository.ts)
+
+Not  sure if there is another way ([Andrei](https://github.com/andrei-tatar) please confirm/deny) but to create the Database Tables I had to modify this lines
```
Comment (andrei-tatar) [inline README.md]: Yes, unfortunately there's no other DB creation/migration code.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,103 @@
...
+        console.error(err);
+    }).then(() => {
+        console.log('done');
+    });
+
+
+## PUSH
+
+I think THAT'S IT, you are ready to deploy to Heroku! 
+I would suggest to build the app and the  schema (just to check we didn't mess up [Andrei's](https://github.com/andrei-tatar) code :grimacing:) with `npm run build && npm run schema` before deploying.
```
Comment (andrei-tatar) [inline README.md]: You can actually run `npm run heroku-postbuild` and it will do the code build, json schema and copy the html files to the dist folder.

**Review by andrei-tatar — COMMENTED** [review]
Answered the questions. Thanks for detailing the setup!

I think heroku needs an extra step to add postgress as a resource.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,103 @@
...
+	- Scopes (I added google-home-authcode / app-user because i saw those 2 somewhere in the code but I don't think they are needed) ?? @Andrei please confirm/deny
+- Back in Overview click Add Actions
+- Fulfillment URL - https://XXXXXXXXX.herokuapp.com/smarthome/fulfill
+
+
+## CONFIG.TS
+
+ Pull the repo, and modify [src/config.ts](https://github.com/andrei-tatar/nora-service/blob/master/src/config.ts)
+
+    export const oauthClientId = 'api console - credentials - oauth 2.0 client ids - web client - Client ID';
```
Comment (andrei-tatar) [inline README.md]: The code doesn't need changing. These values are pulled from environment variables which can be set directly in heroku /settings/config vars

image

**Comment by nicandris** [comment]
fixed the comments. can you take a look at the env vars?

**Comment by andrei-tatar** [comment]
@nicandris Looks good! The `OAUTH_PROJECT_ID` is not really needed. Actually a new firebase project is not really needed because creating a "Smart home" project in Google Actions actually creates a new firebase project. But it should work either way so doesn't really matter. I'll merge as is for now.

**Pull request merged**